daemon/logs.go

```diff
@@ -18,9 +18,16 @@ import (
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
-// ContainerLogs hooks up a container's stdout and stderr streams
-// configured with the given struct.
-func (daemon *Daemon) ContainerLogs(ctx context.Context, containerName string, config *backend.ContainerLogsConfig, started chan struct{}) error {
+// ContainerLogsRawChan, given a channel, returns raw log message structs from
+// the container logger.
+func (daemon *Daemon) ContainerLogsRawChan(ctx context.Context, containerName string, config *backend.ContainerLogsConfig, msgs chan *logger.Message) error {
```

Comment:
There is a good reason to do it like this: it the API server implementation doesn't have to be aware of the details of the logging format, and doesn't have logger as a dependency.